### DISCUSSION TITLE
Modify view of a Settings page via module?

### DISCUSSION THREAD

**Discussion by willnationsdev** [OrchardCMS/OrchardCore#8400] [Q&A] [answered]
I want to add a configurable value to the LoginSettings when my module is enabled and have it independently handle the DisplayDriver for the field to set the value. Is there any way to do this, or am I required to make my own settings be on my own page?

I tried to just get some "test" text in a view to show up on the LoginSettings page by putting my view in the following places, none of which have worked thus far:

- `Views\OrchardCore.Users\LoginSettings.cshtml`
- `Views\OrchardCore.Users\LoginSettings.Edit.cshtml`
- `Views\OrchardCore.Users\Admin\Settings\LoginSettings.cshtml`
- `Views\OrchardCore.Users\Admin\Settings\LoginSettings.Edit.cshtml`
- `Views\Shared\Admin\Settings\LoginSettings.cshtml`
- `Views\Shared\Admin\Settings\LoginSettings.Edit.cshtml`

The docs seem to indicate that you can manipulate existing views by putting them in your theme, but I'm wanting to do it in a typical module. Thoughts?

**Comment by jsalgado10** [comment, marked as answer]
@willnationsdev you could take a look at the LoginSettingsDisplayDriver class on the Users Module

**Reply by willnationsdev** [reply]
As far as I understand, that will only allow me to manipulate the ViewModel for GET and POST requests on the page, or to modify the display of the page on the frontend (if there was one). I can't identify how I actually write out the HTML content for what I want to add to the admin view of the page, without making core changes to the OC repository (want to do it only from a module).

Is it possible to directly modify the admin HTML content from the Edit/Update methods on the DisplayDrivers?

**Reply by jsalgado10** [reply]
This line is return the LoginSettings.Edit view

`` return Initialize<LoginSettings>("LoginSettings_Edit", model =>
            {
                model.UseSiteTheme = settings.UseSiteTheme;
                model.UseExternalProviderIfOnlyOneDefined = settings.UseExternalProviderIfOnlyOneDefined;
                model.DisableLocalLogin = settings.DisableLocalLogin;
                model.UseScriptToSyncRoles = settings.UseScriptToSyncRoles;
                model.SyncRolesScript = settings.SyncRolesScript;
            }).Location("Content:5").OnGroup(GroupId); ``

From what i understand the SectionDisplayDriver will add to the view on a specific groupid. In this case you can create your own displaydriver, ViewModel and View and use GroupId "userLogin" to display and update the site properties correctly on the LoginSettings section

The View will be your Html page.

**Reply by willnationsdev** [reply]
I will have to look into that further then. Thanks!

**Reply by jsalgado10** [reply]
Let me know if you need any help. I will be happy to help
